File: WorldFeed/Server/src/Services/Message/WorldFeed.Message.API/DTO/QueryDTO/Cursor/IMessageCursorQueryResultDTO.cs
namespace WorldFeed.Message.DTO.QueryDTO.Cursor
{
    using System.Collections.Generic;

    using WorldFeed.Common.Public.Models.Interfaces;
    using WorldFeed.Common.Public.Models.Interfaces.DTO.QueryDTO;
    using WorldFeed.Message.DTO.Events;

    public interface IMessageCursorQueryResultDTO : IBaseCursorQueryDTO<IMessageEventDTO>
    {
        IMessageEventDTO[] MessageEvents { get; set; }

        Dictionary<long, IApp> Apps { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Interfaces/DTO/QueryDTO/IMessageCursorQueryResultDTO.ts
import {IBaseCursorQueryDTO} from "./IBaseCursorQueryDTO";
import {IMessageEventDTO} from "../Events/IMessageEventDTO";
import {IApp} from "../../IApp";
import Dictionary from "typescript-dotnet-commonjs/System/Collections/Dictionaries/Dictionary";

export interface IMessageCursorQueryResultDTO extends IBaseCursorQueryDTO<IMessageEventDTO> {
  messageEvents: IMessageEventDTO[];
  apps: Dictionary<number, IApp>;
}
